src/compass/utils/validate_cslc.py

```diff
@@ -110,12 +110,33 @@ def compare_cslc_products(file_ref, file_sec):
         return
 
     # Compare amplitude of reference and generated CSLC products
-    print('Check mean real part difference between CSLC products is < 1.0e-5')
-    assert np.allclose(slc_ref.real, slc_sec.real,
-                       atol=0.0, rtol=1.0e-5, equal_nan=True)
-    print('Check mean imaginary part difference between CSLC products is < 1.0e-5')
-    assert np.allclose(slc_ref.imag, slc_sec.imag,
-                       atol=0.0, rtol=1.0e-5, equal_nan=True)
+    np.seterr(invalid='ignore')
+    diff_real = np.abs((slc_ref.real - slc_sec.real) / slc_sec.real)
+    diff_imag =  np.abs((slc_ref.imag - slc_sec.imag) / slc_sec.imag)
+
+    # Compute total number of pixels different from nan from ref and sec
+    tot_pixels_ref = np.count_nonzero(~np.isnan(np.abs(slc_ref)))
+    tot_pixels_sec = np.count_nonzero(~np.isnan(np.abs(slc_ref)))
+
+    # Check that total number of pixel is the same
+    assert tot_pixels_ref == tot_pixels_sec
```

Comment:
I don't think comparing number of Nan's is a sufficiently robust check. The following toy example of 2 4x4 rasters (red Nan and green valid) would pass, when I think it ought to fail.
![image](https://user-images.githubusercontent.com/2991893/210288812-fad1bf26-2a81-4689-af80-f29ace4f800a.png)

I think following per-pixel test would better validate that geolocation is consistent the same between the rasters:
```python
ref_nan = isnan(slc_ref)
sec_nan = isnan(slc_sec)
np.testing.assert_array_equal(ref_nan, sec_nan)
```

I left out `np.abs` because if either real or imaginary component is NaN, `np.nan` evaluates to `True`. e.g.:
```python
In [4]: np.isnan([1, np.nan + np.nan * 1j, np.nan + 1j, 1 + np.nan * 1j])
Out[4]: array([False,  True,  True,  True])
```

Reply:
uhm, I am a bit confused. My lines of code are comparing pixels that are not NaNs and leaving aside those pixels which are NaNs. With your code, we would compare only NaNs. The idea of the test is that the number of valid pixels (not NaNs) in the reference and secondary RSLCs are the same. Your code would do the opposite, doesn't it?

Reply:
By using `np.count_nonzero`, only the number of non-NaN values is compared and no consideration is given for where the values are located in the array. Both toy rasters have 2 non-zero pixels at different locations and it incorrectly passes the current assert.

`np.testing.assert_array_equal` does an element-by-element comparison of the NaN mask and it would correctly fail with the toy rasters since the non-NaNs are at different places.